### ISSUE TITLE
"human" function sometimes does not format full minutes correctly

### ISSUE THREAD

**Issue by ngollon** [sharkdp/numbat#668]
The "human" function sometimes does not output the given duration in the shortest possible form.

```
human(3720 second)
 -> "1 hour + 2 minutes"
```

which I think is correct, but

```
human(3840s)
 -> "1 hour + 3 minutes + 60 seconds
```

which I think should be `1 hour + 4 minutes`.

**Comment by zxsa89675** [comment]
Hello, we tried to solve the issue.

This is what we did:

Modified the _human_recurse function to round up to the next unit when appropriate, ensuring the human function outputs the duration in the shortest possible form.

You can review changes in this commit: https://github.com/zxsa89675/sharkdp-numbat-668/commit/7f5e498898b04e942fc001ac2d652cf2f63386c0.


> [!CAUTION]
> Disclaimer: The concept of solution was created by AI and you should never copy paste this code before you check the correctness of generated code. Solution might not be complete, you should use this code as an inspiration only.

---

If you no longer want Latta AI to attempt solving issues on your repository, you can block this account.

**Comment by Goju-Ryu** [comment]
I reported this in #622 and have a fix in #661 as a result of reimplementing the human function using mixed units. 
Currently that PR is awaiting review, but I think it might be a bit before it is merged due to the holidays and all.

**Comment by ngollon** [comment]
Sorry, I did not find your ticket. 
This is indeed a duplicate.

Thanks!

**Comment by Goju-Ryu** [comment]
No worries, I just wanted to give the context and link it for future reference :)

**Comment by sharkdp** [comment]
Fixed in v1.15.
